13/root/usr/share/container-scripts/postgresql/common.sh

```diff
@@ -297,7 +297,7 @@ function wait_for_postgresql_master() {
 run_pgupgrade ()
 (
   # Remove .pid file if the file persists after ugly shut down
-  if [ -f "$PGDATA/postmaster.pid" ] && ! pgrep -f "postgres" > /dev/null; then
```

Comment:
Why are you removing the condition? We do not want to remove `.pid` file of  the running process

Reply:
So I removed it because I had some issues when I was testing the upgrade. I will look into it deeper.
we should check if the PID  is up first

Reply:
so this is from a running container I added a stop before the line
seem that the service is not running, also there isn't any open socket

I'm assuming the it is partially match `run-postgresql` 
```
bash-4.4$ pgrep -f "postgres" 
1
17

 ps -ef
UID          PID    PPID  C STIME TTY          TIME CMD
postgres       1       0  0 12:18 pts/0    00:00:00 /bin/bash /usr/bin/run-postgresql
postgres      17       1  0 12:18 pts/0    00:00:00 /bin/bash /usr/bin/run-postgresql
postgres      18      17  0 12:18 pts/0    00:00:00 /usr/bin/coreutils --coreutils-prog-shebang
postgres      19       0  0 12:19 pts/1    00:00:00 bash
postgres      27      19  0 12:22 pts/1    00:00:00 ps -ef
```

how about 
```suggestion
  if [ -f "$PGDATA/postmaster.pid" ] && ! pg_isready > /dev/null; then
 ```

Reply:
changed

Reply:
makes sense +1